### ISSUE TITLE
loop in initialsync

### ISSUE THREAD

**Issue by k-mooijman** [jaetzold/philips-hue-java-sdk#5]
the line `parseGroups(datastore.getJSONObject("groups"));`in completeSync triggers a `getLight` and so a `checkAuthAndSync();` where it checks for a !initialSyncDone and as the initialSyncDone is set after the `parseGroups(datastore.getJSONObject("groups"));` is will initiate a new completeSync creating a infinite loop

**Comment by jaetzold** [comment]
Yes, looks like it. I never really used groups on my bridge. That code looks like it would never work with groups on the bridge already defined?

Try to replace [`final HueLightBulb light = getLight(lightId);`](https://github.com/jaetzold/philips-hue-java-sdk/blob/eff7d72c9916221b3e6e4a05a239ecda88df2684/src/main/java/de/jaetzold/philips/hue/HueBridge.java#L533)
with `final HueLightBulb light = lights.get(lightId);`.
I would be happy to know if that does allow you to sync groups properly.
